### PR TITLE
feat!: upgrade next-metrics

### PR DESCRIPTION
**Pull request by frshwtr**
Replaces n-logger with reliability kit logger.

### CODE REVIEW

**Comment by notlee** [comment]
As this is a major release could you also write a migration guide (MIGRATION.md) please? 🙏 And update the migration table in the README accordingly. There's also mention of node@10 in the README 😱 We can remove the mention of a specific node version there I think.

**Comment by notlee** [comment]
There's a couple of other Github Actions using node@12. Do you think it's worth updating those to totally remove unsupported versions of node from this project as part of this PR?

**Comment by frshwtr** [comment]
> There's a couple of other Github Actions using node@12. Do you think it's worth updating those to totally remove unsupported versions of node from this project as part of this PR?

Absolutely. It's good to have the consistency.

File: README.md [hunk 1]
```diff
@@ -245,14 +289,17 @@ New versions of the module are published automatically by CI when a new tag is c
...
 We also maintain a [migration guide](MIGRATION.md) to help you migrate.
 
 | :grey_question: | Major Version | Last Minor Release | Node.js Versions | Support End Date |
-| :-------------- | :------------ | :----------------- | :--------------- | :--------------- |
-| :heart:         | 5             | N/A                | 10+              | N/A              |
-| :hourglass:     | 4             | 4.1                | 8+               | 2020-12-06       |
+|:----------------|:--------------|:-------------------|:-----------------|:-----------------|
+| :heart:         | 6             | N/A                | 18+              | 2025-04-30       |
+| :hourglass:     | 5             | N/A                | 10+              | 2021-04-30       |
+| :skull:         | 4             | 4.1                | 8+               | 2020-12-06       |
```
Comment (notlee) [inline README.md:302]: @frshwtr this indicates our support, not NodeJS support. Let's drop support pretty quickly this time given this is mostly internal to us, and to align with next-metrics.

```suggestion
| :heart:         | 6             | N/A                | 18+              | N/A       |
| :hourglass:     | 5             | N/A                | 10+              | 2023-09-01       |
| :skull:         | 4             | 4.1                | 8+               | 2020-12-06       |
```

**Review by notlee — APPROVED** [review]

**Pull request merged**